Library/Homebrew/formula.rb

```diff
@@ -980,7 +998,7 @@ def migration_needed?
   end
 
   # @private
-  def outdated_versions
+  def outdated_versions(options = {})
     @outdated_versions ||= begin
```

Comment:
I think a simple solution would be to cache based on the given options, though I assume this will continue to be slightly incorrect, as I think `:fetch_head => true` means “fetch HEAD now (possibly again) and then return a result based on this most recent fetch”.

A cleaner solution would be to think about what doesn't change during the lifetime of a Homebrew process and only cache that. Then refactor the method to compute the final result from the information in the cache and add to that the information that can potentially change on every invocation of this method.